spec/edpm-openstack-service-architecture.md

```diff
@@ -0,0 +1,326 @@
+  readiness can only be collected by checking the Ready condition of every
+  service CRs related to this EDPM node.
+
+
+### Example
+
+NovaCell/nova-cell1
+```yaml
+apiVersion: nova.openstack.org/v1beta1
+kind: NovaCell
+metadata:
+    name: nova-cell1
+...
+```
+No changes in NovaCell.
+
+OpenstackDataPlaneRole/edpm-compute
+```yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneRole
+metadata:
+    name: edpm-compute
+    annotations:
+        nova.openstack.org/cell: nova-cell1
+spec:
+    dataPlane: openstack-edpm
+    nodeTemplate:
+       ansiblePort: 22
+...
+```
+The `novaTemplate` is removed from the `nodeTemplate` and a new user specified
+annotation `nova.openstack.org/cell` is added so the human deployer can
+define that this Role describe a set of EDPM nodes that are compute nodes
+connected to cell1.
+
+OpenstackDataPlaneNode/edpm-compute-0
+```yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneNode
+metadata:
+    name: edpm-compute-0
+spec:
+    ansibleHost: 192.168.122.100
+    hostName: edpm-compute-0
+    node:
+        ansibleSSHPrivateKeySecret: dataplane-ansible-ssh-private-key-secret
+    role: edpm-compute
+...
+```
+The `novaTemplate` is removed from the `node`. As this DataPlaneNode refers
+to a DataPlaneRole that has a `nova.openstack.org/cell` annotation, this
+DataPlaneNode is a compute node connected to cell1.
+
+OpenstackDataPlaneNode/edpm-compute-1
+```yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneNode
+metadata:
+    name: edpm-compute-1
+    annotations:
+        nova.openstack.org/cell: nova-cell2
+spec:
+    ansibleHost: 192.168.122.101
+    hostName: edpm-compute-1
+    node:
+        ansibleSSHPrivateKeySecret: dataplane-ansible-ssh-private-key-secret
+    role: edpm-compute
+...
+```
+This DataPlaneNode has an `nova.openstack.org/cell` annotation that overrides
+the cell annotation of its role. So this EDPM node is a compute node connected
+to cell2.
+
+NovaExternalCompute=edpm-compute-0
+```yaml
+apiVersion: nova.openstack.org/v1beta1
+kind: NovaExternalCompute
+metadata:
+    name: edpm-compute-0
+spec:
+    dataplaneNodeName: edpm-compute-0
+    cellName: nova-cell1
+    customServiceConfig: |
+    # human deployer can set this to pass through service specific config
+    sshKeySecretName: dataplane-ansible-ssh-private-key-secret
+...
+```
+This CR is auto created by the nova-operator based on the corresponding
+DataPlaneNode CR and the cell annotation set on it. The spec fields, except
+`CustomServiceConfig` is populated by the nova-operator. The
+`CustomServiceConfig` field is never set by the nova-operator it can be used
+by the human deployer.
+
+The following fields will be copied over from the DataplaneNode (or Role due to
+inheritance) and changes there will be propagated to the NovaExternalCompute CR
+* InventoryConfigMapName
+* NetworkAttachments
+* SshKeySecretName
+
```

Comment:
its not that it copied persay but its also probably good to note that the 
openstackDataplaneNode name will be used as the name of the NovaExternalCompute

in this case edpm-compute-0

we could technically not have them match and do something like
<nova cell>-<dataplane node> i.e. nova-cell1-edpm-compute-0
but that would d add complexity for little value.

un like the other filed that are copied however the CR name will be immutable.